File: src/my_project/easy_problems/from151To200/check_if_word_equals_summation_two_words.py
from typing import List 

class Solution:
    def isSumEqual(self, firstWord: str, secondWord: str, targetWord: str) -> bool:


        def fromStringToCode(word: str) -> int:
            temp = 0
            len_word = len(word)
            for i in range(len(word)-1,-1,-1):
                # print((ord(word[i])%97)* (10**(i)), word[i])
                temp += (ord(word[i])%97)* (10**(len_word -1- i))


            return temp

        
        
        return fromStringToCode(firstWord) + fromStringToCode(secondWord) == fromStringToCode(targetWord) 

solution = Solution()

print(solution.isSumEqual(firstWord = "acb", secondWord = "cba", targetWord = "cdb"))

print(solution.isSumEqual(firstWord = "aaa", secondWord = "a", targetWord = "aab"))

print(solution.isSumEqual(firstWord = "aaa", secondWord = "a", targetWord = "aaaa"))

sample = 'abc'

for i in range(len(sample)):
    print(sample[i])

File: src/my_project_typescript/easy_problems_typescript/from51To100/checkIfWordEqualsSummationTwoWords.ts
function isSumEqual(firstWord: string, secondWord: string, targetWord: string): boolean {


    function fromStringToNumber(word: string): number{
        let temp: number = 0
        let lengthWord: number = word.length

        for (let i = word.length-1; i > -1; i = i -1){
            temp += (word[i].charCodeAt(0)%97)*(10**(lengthWord-1-i))
        }
        return temp 
    }

    return fromStringToNumber(firstWord) + fromStringToNumber(secondWord) === fromStringToNumber(targetWord)

};

console.log(isSumEqual("acb", "cba",  "cdb"))

console.log(isSumEqual("aaa",  "a", "aab"))

console.log(isSumEqual("aaa",  "a",  "aaaa"))
